tests/unit/test_matrix.py

```diff
@@ -76,3 +76,39 @@ def dimension_(self, request):
     @pytest.fixture
     def _SumSubtotals_(self, request):
         return class_mock(request, "cr.cube.matrix._SumSubtotals")
+
+
+class Describe_BaseCubeResultMatrix(object):
+    @pytest.mark.parametrize(
+        ("is_mr_aug", "has_means", "factory_method"),
+        (
+            (True, None, "_mr_aug_matrix_factory"),
+            (True, True, "_mr_aug_matrix_factory"),
+            (True, False, "_mr_aug_matrix_factory"),
+            (False, True, "_means_matrix_factory"),
+            (None, True, "_means_matrix_factory"),
+            (False, False, "_regular_matrix_factory"),
+            (None, False, "_regular_matrix_factory"),
+            (False, None, "_regular_matrix_factory"),
+        ),
+    )
+    def it_calls_the_correct_factory_method_for_appropriate_matrix_type(
+        self, request, cube_, is_mr_aug, has_means, factory_method
+    ):
+        # --- set cube properties that dictate logic
+        cube_.is_mr_aug = is_mr_aug
+        cube_.has_means = has_means
+
+        # --- create factory method mock
+        factory = method_mock(request, _BaseCubeResultMatrix, factory_method)
+        factory.return_value = "zoom"
```

Comment:
Use `_BaseCubeResultMatrix` object instead, as an instance mock (name it `cube_result_matrix_`).